gpiozero/pins/pigpiod.py

```diff
@@ -98,7 +99,7 @@ class PiGPIOPin(Pin):
     GPIO_PULL_UP_NAMES = {v: k for (k, v) in GPIO_PULL_UPS.items()}
     GPIO_EDGES_NAMES = {v: k for (k, v) in GPIO_EDGES.items()}
 
-    def __new__(cls, number, host='localhost', port=8888):
+    def __new__(cls, number, host = os.getenv("PIGPIO_ADDR", 'localhost'), port = os.getenv("PIGPIO_PORT", 8888) ):
```

Comment:
Actually those will be evaluated at module import time; function and method defaults are evaluated at declaration time (effectively module import time) rather than execution time. This is one of the reasons that things like this fail in interesting ways:

```python
def my_func(some_param=[]):
    some_param.append(1)
    return some_param

print(my_func())
print(my_func())
print(my_func())
```